pool/static_pool/pool.go

```diff
@@ -201,7 +201,7 @@ begin:
 		rsp, err = w.Exec(context.Background(), p)
 	}
 
-	if sp.cfg.MaxJobs != 0 && w.State().NumExecs() >= sp.cfg.MaxJobs {
+	if sp.cfg.MaxJobs != 0 && w.MaxJobsReached() {
```

Comment:
No need to create an additional method calls since this is a hot place. I'm not sure, that it'll be inlined.